### PR TITLE
feat: upgrade glog to 0.6.0

### PR DESCRIPTION
**Pull request by dl239**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

### CODE REVIEW

File: third-party/cmake/FetchGlog.cmake [hunk 1]
```diff
@@ -12,22 +12,21 @@
...
   DOWNLOAD_DIR ${DEPS_DOWNLOAD_DIR}/glog
   INSTALL_DIR ${DEPS_INSTALL_DIR}
   DEPENDS gflags
   BUILD_IN_SOURCE TRUE
-  CONFIGURE_COMMAND
-    ./autogen.sh
-    COMMAND CXXFLAGS=-fPIC ./configure --prefix=<INSTALL_DIR> --enable-shared=no --with-gflags=<INSTALL_DIR>
-  BUILD_COMMAND ${MAKE_EXE}
-  INSTALL_COMMAND ${MAKE_EXE} install)
+  CONFIGURE_COMMAND ${CMAKE_COMMAND} -H<SOURCE_DIR> -B <BINARY_DIR> -DWITH_GLOG=ON -DCMAKE_CXX_FLAGS=-fPIC
```
Comment (aceforeverd) [inline third-party/cmake/FetchGlog.cmake]: although its the same, `${DPES_INSTALL_DIR}` -> `<INSTALL_DIR>` looks more unified.

File: third-party/cmake/FetchGlog.cmake [hunk 2]
```diff
@@ -12,22 +12,21 @@
...
+    -DCMAKE_PREFIX_PATH=${DEPS_INSTALL_DIR} -DCMAKE_INSTALL_PREFIX=${DEPS_INSTALL_DIR}
+  BUILD_COMMAND ${CMAKE_COMMAND} --build <BINARY_DIR>
```
Comment (aceforeverd) [inline third-party/cmake/FetchGlog.cmake]: inherit the MAKEOPTS options, add those suffix : `-- ${MAKEOPTS}`

**Comment by aceforeverd** [comment]
thirdparty testing working https://github.com/4paradigm/hybridsql-asserts/actions/runs/7543169974, you can download the uploaded artifacts

**Comment by dl239** [comment]
> https://github.com/4paradigm/hybridsql-asserts/actions/runs/7543169974

https://github.com/dl239/hybridsql-asserts/pull/1

**Review by aceforeverd — APPROVED** [review]

**Comment by aceforeverd** [comment]
https://github.com/4paradigm/hybridsql-asserts/actions/runs/7579146429

**Review by vagetablechicken — APPROVED** [review]

**Pull request merged**